xmpp.go

```diff
@@ -181,6 +181,9 @@ type Options struct {
 	// provided to the server as the xmlns:auth attribute of the OAuth2 authentication request.
 	OAuthXmlNs string
 
+	// AuthExternal will be chosen if this is set to true and if server supports it
+	AuthExternal bool
```

Comment:
I think `Mechanism` could also be used for setting auth to `EXTERNAL`: https://github.com/xmppo/go-xmpp/blob/master/xmpp.go#L234